### PR TITLE
Adopting a code formatting standard

### LINKED ISSUE #501: Adopt a code formatting standard
We discussed the adoption of a code formatting standard, using for instance `black`.

## Implications for feature branches

Formatting would mean a 1-time black-ify of the whole code base and one commit. If this is done on `delevop` then all maintainers of feature branches must also black-format all code in their branch before merges happen in the future.

@franzpoeschel suggested a rebase workflow using https://github.com/franzpoeschel/reformat-rebase, which was used to format https://github.com/openPMD/openPMD-api.

Another merge-based workflow could be (0) format `develop` and commit, then sync with formatted `delevop` by (1) formatting code in `feature-branch`

```sh
$ git checkout feature-branch
$ find -name "*.py" | xargs black
```

and (2) merging `develop` into `feature-branch`.

```sh
$ git checkout feature-branch
$ git merge develop
```

This won't alter `delevop` and only add a merge commit to `feature-branch`. It makes sure that `feature-branch` is in sync with `develop`. Any further development in `feature-branch` can later be merged into `develop` as usual.

```sh
$ git checkout develop
$ git merge feature-branch
```

## Tools

As an alternative to `black`, one could use the now popular `ruff` linter, which has a [`ruff format`](https://docs.astral.sh/ruff/formatter/) command. It is back-compatible, faster and offers some more features such as doc string code formatting. But then again, `black` is more stable while  `ruff` is more actively developed.

### PR DESCRIPTION
**Pull request by RandomDefaultUser**
This PR lays a foundation for having a proper code formatting standard within MALA. It employs `black`

It

- [x] provides a basic configuration of `black`
  -  This also introduces a `pyproject.toml` file; it currently does not replace the `setup.py`, but complements it; I would postpone fully replacing it until the upcoming hackathon
- [x] "blackifies" the main code base (including examples and tests)
- [x] Provides a note in the documentation on using `black`
- [x] Gets rid of unused imports (not strictly a code formatting thing, but while I was at it...)

It does not implement automatic code formatting checks. As mentioned in #501 we can setup a CI for this, but I would also postpone this until the upcoming hackathon, when we wanted to tackle the CI anyway. 

Closes #501 (except for the CI, which we will add with the coming CI improvements) and #507.

### CODE REVIEW

**Comment by RandomDefaultUser** [comment]
Ah OK, so I think the `pyproject.toml` and `setup.py` did not completely work together yet. But according to this: https://packaging.python.org/en/latest/guides/modernize-setup-py-project/ they should work. I will investigate!

**Review by elcorto — COMMENTED** [review]
I made small adjustment sugestions in `docs/sourc[docs/source/CONTRIBUTE.md`, else this looks good vis-à-vis formatting. Thanks. Please see my other comment re. unused imports.

**Comment by elcorto** [comment]
Regarding #507, I found 3 left-over ones:

```sh
$ ruff check --select F401 $(find mala -name "*.py") | grep -v __init__
mala/descriptors/atomic_density.py:136:32: F401 [*] `lammps.lammps` imported but unused
mala/descriptors/bispectrum.py:141:32: F401 [*] `lammps.lammps` imported but unused
mala/descriptors/minterpy_descriptors.py:95:32: F401 [*] `lammps.lammps` imported but unused
```

All have the form

```py
        # For version compatibility; older lammps versions (the serial version
        # we still use on some machines) have these constants as part of the
        # general LAMMPS import.
        try:
            from lammps import constants as lammps_constants
        except ImportError:
            from lammps import lammps
```

but the fallback module `lammps` is never used in the code. Given the comment, I *suspect* this to work:

```py
        try:
            from lammps import constants as lammps_constants
        except ImportError:
            from lammps import lammps as lammps_constants
```

File: docs/source/CONTRIBUTE.md [hunk 1]
```diff
@@ -90,6 +90,15 @@ the core development team.
 * If you're adding code that should be tested, add tests
 * If you're adding or modifying examples, make sure to add them to `test_examples.py`
 
+### Formatting code
+
+* MALA uses `black` for code for unified code formatting
```
[the remark this answers is not in the document]
Comment (RandomDefaultUser) [reply docs/source/CONTRIBUTE.md:95]: Sounds good, will use this!

File: docs/source/CONTRIBUTE.md [hunk 2]
```diff
@@ -90,6 +90,15 @@ the core development team.
...
+  * For more info on `black` itself, see the respective 
+    [documentation](https://github.com/psf/black)
```
[the remark this answers is not in the document]
Comment (RandomDefaultUser) [reply docs/source/CONTRIBUTE.md:98]: Goes with the other suggestion.

**Comment by RandomDefaultUser** [comment]
You're right, there are three leftovers. I consciously left them in, because I am not entirely sure myself anymore how the old version handled `lammps_constants`. Looking at your suggestion and reflecting at what happens there made me realize there is no point in being compatible here. A LAMMPS version that old will not work with our current codebase anymore. 
I will delete these lines tomorrow, and then all unused imports should be gone.

**Review by elcorto — APPROVED** [review]
Thanks, looks all good now.

**Pull request merged**